synapse/rest/client/v1/directory.py

```diff
@@ -159,7 +159,7 @@ def __init__(self, hs):
     def on_GET(self, request, room_id):
         room = yield self.store.get_room(room_id)
         if room is None:
-            raise SynapseError(400, "Unknown room")
+            raise SynapseError(404, "Unknown room")
```

Comment:
ok this looks good to me, but the error code will still be `M_UNKNOWN`. Really it ought to be `M_NOT_FOUND` as per https://matrix.org/docs/spec/client_server/unstable.html#get-matrix-client-r0-directory-list-room-roomid.

I think that there's a `NotFoundError` or something you can raise instead?

Reply:
Fixed this in the latest commit to raise `NotFoundError` instead of the generic one.